numpy/random/mtrand/mtrand.pyx

```diff
@@ -4491,12 +4496,19 @@ cdef class RandomState:
         # not zero. We continue to use the SVD rather than Cholesky in
         # order to preserve current outputs. Note that symmetry has not
         # been checked.
+
         (u, s, v) = svd(cov)
-        neg = (np.sum(u.T * v, axis=1) < 0) & (s > 0)
-        if np.any(neg):
-            s[neg] = 0.
-            warnings.warn("covariance is not positive-semidefinite.",
-                          RuntimeWarning)
+
+        if check_valid != 'ignore':
+            psd = np.allclose(np.dot(v.T * s, v), cov)
```

Comment:
I realize. I wonder if tol should be changed to atol and rtol and just forward them in here.